springwolf-plugins/springwolf-kafka-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/KafkaPayloadTypeExtractor.java

```diff
@@ -0,0 +1,78 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.channels;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.messaging.handler.annotation.Payload;
+import sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class KafkaPayloadTypeExtractor {
```

Comment:
The ampq plugin also has this exact code. Maybe we should catch this opportunity to use this new class for the amqp scanner as well? The payload type can be a list there as well (and also for non-batch kafka,=)

Reply:
The payload extractor was moved to core now (requiring the spring-messaging dependency in core as well)